gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -310,14 +310,16 @@ private Map<String, BoundField> getBoundFields(Gson context, TypeToken<?> type,
 
   static abstract class BoundField {
     final String name;
+    final Field field;
```

Comment:
As side note: Most likely `BoundField` (or rather its anonymous subclass) already had an implicit `Field` field since it captured the `Field` from the enclosing scope:
https://github.com/google/gson/blob/bc04dd20b969cb8a6999282b6749266b9f477bcf/gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java#L164-L170